BootloaderCommonPkg/Library/ExtraBaseLib/Ia32/CpuSupport.nasm

```diff
@@ -57,6 +57,10 @@ FlushExit:
 ;------------------------------------------------------------------------------
 global ASM_PFX(AsmEnableAvx)
 ASM_PFX(AsmEnableAvx):
+    push    eax

```

Comment:
According to calling convention,  EAX, ECX, EDX is not required to be preserved.
And EBX is not used here.    Want to understand why it is necessary.

Reply:
Sorry,  you are right ,  EBX will be used by CPUID.  
But the code will just need to push/pop EBX.   No need to do it for EAX, ECX and EDX.

Reply:
This is gcc build. I need to go back and check gcc calling convention, but this is to save/restore used registers regardless of calling convention.

Reply:
cdecl calling convention should be the same on gcc.  
So only EBX needs to be saved.

Reply:
Yes, you are right. Out tool chain is using cdecl calling convention. The eax, ecx and edx are scratch registers, so we don't have to preserve them. Let me save/restore ebx only.